raiden/tests/unit/fuzz/test_state_changes.py

```diff
@@ -0,0 +1,442 @@
+
+            self.channels.append(channel)
+
+    def event(self, description):
+        """ Wrapper for hypothesis' event function.
+
+        hypothesis.event raises an exception when invoked outside of hypothesis
+        context, so skip it when we are replaying a failed path.
+        """
+        if not self.replay_path:
+            event(description)
+
+    @precondition(lambda self: self.channels)
+    @invariant()
+    def channel_state_invariants(self):
+        for netting_channel in self.channels:
+            our_state = netting_channel.our_state
+            partner_state = netting_channel.partner_state
+
+            our_balance = channel.get_balance(our_state, partner_state)
+            partner_balance = channel.get_balance(partner_state, our_state)
+
+            assert channel.get_amount_locked(our_state) <= our_balance
+            assert channel.get_amount_locked(partner_state) <= partner_balance
+
+
+class InitiatorState(ChainStateStateMachine):
+
+    def __init__(self):
+        super().__init__()
+        self.invalidated_transfer_ids = set()
+        self.initiated = set()
+
+        self.failing_path_2 = False
+
+    @property
+    def channel(self):
+        return self.channels[0]
+
+    def _action_init_initiator(self, transfer: TransferDescriptionWithSecretState):
+        return ActionInitInitiator(
+            transfer,
+            [factories.route_from_channel(self.channel)],
+        )
+
+    def _receive_secret_request(self, transfer: TransferDescriptionWithSecretState):
+        secrethash = sha3(transfer.secret)
+        return ReceiveSecretRequest(
+            payment_identifier=transfer.payment_identifier,
+            amount=transfer.amount,
+            expiration=self.block_number + 10,  # todo
+            secrethash=secrethash,
+            sender=transfer.target,
+        )
+
+    transfers = Bundle('transfers')
+    init_initiators = Bundle('init_initiators')
+    invalid_authentic_secret_requests = Bundle('invalid_authentic_secret_requests')
+    unauthentic_secret_requests = Bundle('unauthentic_secret_requests')
+
+    @rule(
+        target=transfers,
+        payment_id=integers(min_value=1),
+        amount=integers(min_value=1, max_value=100),
+        secret=secret(),
+    )
+    def populate_transfer_descriptions(self, payment_id, amount, secret):
+        return TransferDescriptionWithSecretState(
+            payment_network_identifier=self.payment_network_id,
+            payment_identifier=payment_id,
+            amount=amount,
+            token_network_identifier=self.token_network_id,
+            initiator=self.address,
+            target=self.channel.partner_state.address,
+            secret=secret,
+        )
+
+    def _secret_in_use(self, secret):
+        return sha3(secret) in self.chain_state.payment_mapping.secrethashes_to_task
+
+    def _available_amount(self):
+        return channel.get_distributable(self.channel.our_state, self.channel.partner_state)
```

Comment:
I believe, `channel.get_distributable...` is missing a reference to `self` :)

Reply:
I don't think so. `get_distributable` is a function in the `raiden.transfer.channel` module, not a method of `self.channel`

Reply:
> Maybe consider direct import of the raiden.transfer.channel methods in use, to help the 

This is how we use the `channel` functions of the state machine all over the codebase.